tsnet/tsnet.go

```diff
@@ -760,6 +760,95 @@ func (s *Server) APIClient() (*tailscale.Client, error) {
 // Listen announces only on the Tailscale network.
 // It will start the server if it has not been started yet.
 func (s *Server) Listen(network, addr string) (net.Listener, error) {
+	return s.listen(network, addr, listenOnTailnet)
+}
+
+// FunnelOption is an option passed to ListenFunnel to configure the listener.
+type FunnelOption string
```

Comment:
I'm concerned that `string` is going to corner us in the future.

Reply:
I don't really see how, but what alternative would you prefer?

Reply:
My concern only concern is that we won't be able to pass parameters.

We can do the usual modifier function option thing.

Like

`type ListenOption func(*listenConfig)`

Reply:
I went with the most generic of the option type variants, figuring it gives us the flexibility to convert later.

Reply:
Thanks!